index.js

```diff
@@ -0,0 +1,135 @@
+const ALL_TRIPS = 'https://trektravel.herokuapp.com/trips';
+
+
+// ~~~~~~~~ERROR & STATUS REPORTING~~~~~~~~
+const reportStatus = (message) => {
+  const statusContainer = $('#status-message');
+  statusContainer.empty();
+  statusContainer.append(`<p>${message}</p>`);
+}
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += '</ul>';
+  reportStatus(content);
+};
+
+// ~~~~~~~~LOAD ALL TRIPS (got from Ada Pets)~~~~~~~~
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trips-list');
+  tripList.empty();
+
+  axios.get(ALL_TRIPS)
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+
+
+      response.data.forEach((trip) => {
+        tripList.append(`<li class='${trip.id}'> ${trip.name}</li>`);
+      })
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+
+      console.log(error);
+    });
+}
+
+//~~~~~~~~TRIP DETAILS~~~~~~~~
+const tripDetails = (id) => {
+  reportStatus('Loading trip details...');
+
+  const tripInfo = $('#trip-info');
+  tripInfo.empty();
+
+  axios.get(ALL_TRIPS + '/' + id)
+    .then((response) => {
+      reportStatus(`Successfully loaded details about ${id}.`);
+
+      let trip = response.data;
+
+      tripInfo.append(`<h3>Name: ${trip.name}</h3>`);
+      tripInfo.append(`<p><strong>Continent: </strong>${trip.continent}</p>`);
+      tripInfo.append(`<p><strong>Category: </strong>${trip.category}</p>`);
+      tripInfo.append(`<p><strong>Weeks: </strong>${trip.weeks}</p>`);
+      tripInfo.append(`<p><strong>Cost: </strong>$${trip.cost}</p>`);
+      tripInfo.append(`<p><strong>About:</strong> ${trip.about}</p>`);
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trip: ${error.message}`);
+      console.log(error);
+    });
+}
+
+//~~~~~~~~TRIP RESERVATION (Ada Pets)~~~~~~~~
+
+//~~~~~~~~Form Data~~~~~~~~
+const readFormData = () => {
+  const parsedFormData = {};
+
+  const nameFromForm = $(`#reservation-form input[name="name"]`).val();
+  parsedFormData['name'] = nameFromForm ? nameFromForm : undefined;
+
+  const emailFromForm = $(`#reservation-form input[name="email"]`).val();
+  parsedFormData['email'] = emailFromForm ? emailFromForm : undefined;
+
+  return parsedFormData;
+};
+
+//~~~~~~~~Clear Form~~~~~~~~
+const clearForm = () => {
+  $(`#reservation-form input[name="name]`).val('');
+  $(`#reservation-form input[name="email]`).val('');
+}
+
+//~~~~~~~~Reserve Trip~~~~~~~~
+const reserveTrip = (event) => {
+  event.preventDefault();
+
+  const tripData = readFormData();
+  console.log(tripData);
+
+  reportStatus('Sending trip data...');
+
+  let tripId = $(this).html();
```

Comment:
Your event handler is an **arrow function** so... it doesn't have a `this` reference. You could use `event.target` instead.  But of course you'll need something in this method to indicate which trip to submit.  Maybe you could put it in the form as a hidden field, or you could use a closure.

Reply:
Post request is now complete.